axlearn/common/measurement.py

```diff
@@ -15,30 +16,20 @@ class Event(enum.Enum):
     """Event to be recorded.
 
     Attributes:
-        START_JOB: Start of job.
```

Comment:
Out of curiosity, I'm wondering whether it makes sense to introduce such breaking changes without notice since we have experiments that are relying on the current API.

One option could be to improve the good put recorder implementation first and then introduce the breaking changes after?

Since axlearn is an open-source project, it might also impact other users outside of Apple?

Reply:
That's not the case actually, unless the `axlearn/common/measurement.py` changes are unintentional.

What's happening here is that if another client builds on top of the recorder api (e.g implementations a custom recorder), then the new axlearn version will start breaking.

I discussed this with @Ethanlm internally and he can fill you in on the context.

Right now this change will break internal experiments which use a different recorder than the good put recorder although the different recorder implements the same API.

My suggestion at the moment would be to only change the good put recorder but not break the API

Reply:
Thanks for the context and discussions on the internal code-base dependencies, I wasn't aware of these (there is no way for me to test them either.)
The global recorder should now be completely backward compatible, it adds a new monitoring api `maybe_monitor_all_goodput` which is currently only implemented by the GoodputRecorder version in gcp/..